src/plots/plots.js

```diff
@@ -287,17 +287,26 @@ plots.resize = function(gd) {
         if(gd._redrawTimer) clearTimeout(gd._redrawTimer);
 
         gd._redrawTimer = setTimeout(function() {
-            if((gd._fullLayout || {}).autosize) {
-                // autosizing doesn't count as a change that needs saving
-                var oldchanged = gd.changed;
+            // return if there is nothing to resize
+            if(gd.layout.width && gd.layout.height) {
+                resolve(gd);
+                return;
+            }
+
+            delete gd._fullLayout._initialAutoSizeIsDone;
+            if(!gd.layout.width) delete (gd._fullLayout || {}).width;
+            if(!gd.layout.height) delete (gd._fullLayout || {}).height;
 
-                // nor should it be included in the undo queue
-                gd.autoplay = true;
+            // autosizing doesn't count as a change that needs saving
+            var oldchanged = gd.changed;
 
-                Plotly.relayout(gd, { autosize: true });
+            // nor should it be included in the undo queue
+            gd.autoplay = true;
+
+            Plotly.plot(gd).then(function() {
```

Comment:
@n-riesco could you clarify why `Plots.resize` needs to go through `Plotly.plot` instead of `Plotly.relayout` now?

I think it has to do with `fullLayout._initialAutoSizeIsDone`, but I just want to make sure.